src/auth/Readme.md

```diff
@@ -0,0 +1,266 @@
+
+```javascript
+  useAuthChange({
+    onSignedOut: () => { /* client invalidate some queries */},
+    onSignedIn: () => { /* client invalidate some queries  */}
+    },
+  });
+```
+
+## How do I sign in?
+
+#### **`SaleorAuthClient.signIn: ({ email, password }: TokenCreateVariables) => Promise<TokenCreateResponse>`**
+
+SaleorAuthClient returns a `signIn` method. You can also access it via the **useSaleorAuthContext** hook.
+
+```javascript
+const { signIn } = useSaleorAuthContext();
+
+const response = await signIn({
+  email: "example@mail.com",
+  passowrd: "password",
+});
+```
+
+## How do I sign out?
+
+#### **`SaleorAuthClient.signOut: () => void`**
+
+This method will remove access and refresh tokens from SaleorAuthClient state and storage. It'll also trigger sign out event, which in turn will trigger [onSignOut method of useAuthChange hook](#how-do-i-tell-my-graphql-client-to-refresh-queries-on-signin--signout)
+
+```javascript
+const { signOut } = useSaleorAuthContext();
+
+signOut();
+```
+
+⚠️ For checkout sign out see [signing out in checkout](#how-do-i-sign-out-in-checkout)
+
+## How do I sign out in checkout?
+
+#### **`SaleorAuthClient.checkoutSignOut: ({ checkoutId }: CustomerDetachVariables) => Promise<CustomerDetachResponse>`**
+
+On top of the regular sign out login, in checkout we need to start signing out process with detaching customer from checkout. Since detach requires a signed in user, it'll happen first and removing tokens from state / storage will only happen if the mutation returned success:
+
+```javascript
+const { checkoutSignOut } = useSaleorAuthContext();
+
+const response = await checkoutSignOut({ checkoutId: checkout.id });
+```
+
+## How does auth handle resetting password?
+
+#### **`SaleorAuthClient.resetPassword: ({ email, password, token }: PasswordResetVariables) => Promise<PasswordResetResponse>`**
+
+SaleorAuthClient class provides you with a reset password method. If the reset password mutation is successful, it'll log you in automatically just like after a regular sign in. The [onSignIn method of useAuthChange hook](#how-do-i-tell-my-graphql-client-to-refresh-queries-on-signin--signout) will also be triggered.
+
+```javascript
+const { resetPassword } = useSaleorAuthContext();
+
+const response = await resetPassword({
+  email: "example@mail.com",
+  password: "newPassword",
+  token: "apiToken",
+});
+```
+
+## How do I use this with Urql?
+
+Once an authentication change happens, you might want to refetch some of your queries. Because Urql doesn't provide a direct way to invalidate cache manually, we're following urql's [proposed approach](https://github.com/urql-graphql/urql/issues/297#issuecomment-501646761) of installing a new instance of the client in place of the old one. We have a hook for that called `useUrqlClient` that takes Urql `ClientOptions` as an only argument and returns the current `urqlClient` and `resetClient` function:
```

Comment:
```suggestion
Once an authentication change happens, you might want to refetch some of your queries. Because urql doesn't provide a direct way to invalidate cache manually, we're following urql's [proposed approach](https://github.com/urql-graphql/urql/issues/297#issuecomment-501646761) of installing a new instance of the client in place of the old one. We have a hook for that called `useUrqlClient` that takes urql `ClientOptions` as the only argument and returns the current `urqlClient` and the `resetClient` function:
```